### ISSUE TITLE
Error while compiling iOS app with google_sign_in (or any other dependency)

### ISSUE THREAD

**Issue by kalehv** [flutter/flutter#14953]
## Steps to Reproduce

Follow the steps to integrate Google Sign In to iOS app from [here](https://github.com/flutter/plugins/tree/master/packages/google_sign_in)

## Logs
`flutter run` output
```
➜  flutter_todo flutter run -d iP8+P
Launching lib/main.dart on iP8+P in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: S29F39HXPA
Running pod install...                                1.3s
Running Xcode clean...                                1.4s
Starting Xcode build...
 ├─Building Dart code...                       3.4s
 ├─Assembling Flutter resources...             4.1s
 └─Compiling, linking and signing...           0.7s
Xcode build done                                     15.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    === BUILD TARGET google_sign_in OF PROJECT Pods WITH CONFIGURATION Debug ===
    /Users/kalehv/.pub-cache/hosted/pub.dartlang.org/google_sign_in-2.1.0/ios/Classes/GoogleSignInPlugin.m:111:41: warning: 'UIApplicationOpenURLOptionsSourceApplicationKey' is only available on iOS 9.0 or newer [-Wunguarded-availability]
      NSString *sourceApplication = options[UIApplicationOpenURLOptionsSourceApplicationKey];
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In module 'UIKit' imported from /Users/kalehv/dev/flutter_todo/ios/Pods/Target Support Files/google_sign_in/google_sign_in-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:534:51: note: 'UIApplicationOpenURLOptionsSourceApplicationKey' has been explicitly marked partial here
    UIKIT_EXTERN UIApplicationOpenURLOptionsKey const UIApplicationOpenURLOptionsSourceApplicationKey NS_AVAILABLE_IOS(9_0);   // value is an NSString containing the bundle ID of the originating application
                                                      ^
    /Users/kalehv/.pub-cache/hosted/pub.dartlang.org/google_sign_in-2.1.0/ios/Classes/GoogleSignInPlugin.m:111:41: note: enclose 'UIApplicationOpenURLOptionsSourceApplicationKey' in an @available check to silence this warning
      NSString *sourceApplication = options[UIApplicationOpenURLOptionsSourceApplicationKey];
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/kalehv/.pub-cache/hosted/pub.dartlang.org/google_sign_in-2.1.0/ios/Classes/GoogleSignInPlugin.m:112:27: warning: 'UIApplicationOpenURLOptionsAnnotationKey' is only available on iOS 9.0 or newer [-Wunguarded-availability]
      id annotation = options[UIApplicationOpenURLOptionsAnnotationKey];
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In module 'UIKit' imported from /Users/kalehv/dev/flutter_todo/ios/Pods/Target Support Files/google_sign_in/google_sign_in-prefix.pch:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:535:51: note: 'UIApplicationOpenURLOptionsAnnotationKey' has been explicitly marked partial here
    UIKIT_EXTERN UIApplicationOpenURLOptionsKey const UIApplicationOpenURLOptionsAnnotationKey NS_AVAILABLE_IOS(9_0);   // value is a property-list typed object corresponding to what the originating application passed in UIDocumentInteractionController's annotation property
                                                      ^
    /Users/kalehv/.pub-cache/hosted/pub.dartlang.org/google_sign_in-2.1.0/ios/Classes/GoogleSignInPlugin.m:112:27: note: enclose 'UIApplicationOpenURLOptionsAnnotationKey' in an @available check to silence this warning
      id annotation = options[UIApplicationOpenURLOptionsAnnotationKey];
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    2 warnings generated.
    === BUILD TARGET google_sign_in OF PROJECT Pods WITH CONFIGURATION Debug ===
    ld: warning: -undefined dynamic_lookup is deprecated on iOS
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    The use of Swift 3 @objc inference in Swift 4 mode is deprecated. Please address deprecated @objc inference warnings, test your code with “Use of deprecated Swift 3 @objc inference” logging enabled, and then disable inference by changing the "Swift 3 @objc Inference" build setting to "Default" for the "Runner" target.
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    /Users/kalehv/dev/flutter_todo/ios/Runner/GeneratedPluginRegistrant.m:6:9: fatal error: 'google_sign_in/GoogleSignInPlugin.h' file not found
    #import <google_sign_in/GoogleSignInPlugin.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
Could not build the precompiled application for the device.

Error launching application on iP8+P.
```
---
`flutter analyze` output
```
➜  flutter_todo flutter analyze
Analyzing /Users/kalehv/dev/flutter_todo...
No issues found!
Ran in 4.9s
```

## Flutter Doctor

`flutter doctor -v` output (Device IDs redacted)
Note - I am not using IntelliJ IDEA. I am using Android Studio instead.
```
➜  flutter_todo flutter doctor -v
[✓] Flutter (on Mac OS X 10.13.3 17D102, locale en-US, channel beta)
    • Flutter version 0.1.4 at /Users/kalehv/Library/flutter
    • Framework revision f914e701c5 (8 days ago), 2018-02-19 21:12:17 +0000
    • Engine revision 13cf22c284
    • Dart version 2.0.0-dev.27.0-flutter-0d5cf900b0

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/kalehv/Library/Android/sdk
    • Android NDK at /Users/kalehv/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /Users/kalehv/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.0 Preview.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[!] IntelliJ IDEA Community Edition (version 2017.3.4)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.20.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.9.0

[✓] Connected devices
    • iP8+P    • XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX • ios • iOS 11.2.5
    • iPhone X • XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXX     • ios • iOS 11.2 (simulator)

! Doctor found issues in 1 category.
```

As you can see, for some reason, I am not able to add any dependency package and run the app for iOS. This issue is not specific to only google_sign_in. 

Working fine on Android.

<img width="1920" alt="screen shot 2018-02-28 at 12 17 36 am" src="https://user-images.githubusercontent.com/588703/36778576-f59e0620-1c21-11e8-96e4-6735a14a7b48.png">

**Comment by mravn-google** [comment]
Once you've added a Flutter plugin, you've also added a CocoaPods pod to your Xcode project. That means you need to open `ios/Runner.xcworkspace` rather than `ios/Runner.xcodeproj` in Xcode.

**Comment by mravn-google** [comment]
@kalehv Can you run a plugin project on iOS:
```
flutter create -t plugin hello
cd hello/example
flutter run
```

**Comment by kalehv** [comment]
@mravn-google Yup, I know how cocoapods work. I opened `xcworkspace` and not `xcodeproj`
Plugin project worked
```
➜  example flutter run
Launching lib/main.dart on iPhone X in debug mode...
Running pod install...                                1.2s
Running Xcode clean...                                1.2s
Starting Xcode build...
 ├─Assembling Flutter resources...             4.4s
 └─Compiling, linking and signing...           8.5s
Xcode build done                                     19.1s
Syncing files to device iPhone X...                   2.0s

🔥  To hot reload your app on the fly, press "r". To restart the app entirely, press "R".
An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:8100/
For a more detailed help message, press "h". To quit, press "q".
```
I am attaching a screenshot of Pods project. Please have a look at the structure. Does it look like how it should be?
<img width="1920" alt="screen shot 2018-03-01 at 8 04 36 am" src="https://user-images.githubusercontent.com/588703/36855030-6788181a-1d27-11e8-93d1-66ba9d6bd4cf.png">

**Comment by kalehv** [comment]
Okay so looks like my Podfile was a problem (which I did not create)

This is the generated Podfile that does not work
``` ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

def parse_KV_file(file, separator='=')
  file_abs_path = File.expand_path(file)
  if !File.exists? file_abs_path
    return [];
  end
  pods_ary = []
  skip_line_start_symbols = ["#", "/"]
  File.foreach(file_abs_path) { |line|
      next if skip_line_start_symbols.any? { |symbol| line =~ /^\s*#{symbol}/ }
      plugin = line.split(pattern=separator)
      if plugin.length == 2
        podname = plugin[0].strip()
        path = plugin[1].strip()
        podpath = File.expand_path("#{path}", file_abs_path)
        pods_ary.push({:name => podname, :path => podpath});
      else
        puts "Invalid plugin specification: #{line}"
      end
  }
  return pods_ary
end

target 'Runner' do
  use_frameworks!

  # Prepare symlinks folder. We use symlinks to avoid having Podfile.lock
  # referring to absolute paths on developers' machines.
  system('rm -rf Pods/.symlinks')
  system('mkdir -p Pods/.symlinks/flutter')
  system('mkdir -p Pods/.symlinks/plugins')

  # Flutter Pods
  generated_xcode_build_settings = parse_KV_file('./Flutter/Generated.xcconfig')
  if generated_xcode_build_settings.empty?
    puts "Generated.xcconfig must exist. If you're running pod install manually, make sure flutter packages get is executed first."
  end
  generated_xcode_build_settings.map { |p|
    if p[:name] == 'FLUTTER_FRAMEWORK_DIR'
      symlink = File.join('Pods', '.symlinks', 'flutter', File.basename(p[:path]))
      File.symlink(p[:path], symlink)
      pod 'Flutter', :path => symlink
    end
  }

  # Plugin Pods
  plugin_pods = parse_KV_file('../.flutter-plugins')
  plugin_pods.map { |p|
    symlink = File.join('Pods', '.symlinks', 'plugins', File.basename(p[:path]))
    File.symlink(p[:path], symlink)
    pod p[:name], :path => File.join(symlink, 'ios')
  }
end

pre_install do |installer|
  # workaround for https://github.com/CocoaPods/CocoaPods/issues/3289
  Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
    end
  end
end
```

This is the podfile I got from one of the Google Codelabs sample app.
``` ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

if ENV['FLUTTER_FRAMEWORK_DIR'] == nil
    abort('Please set FLUTTER_FRAMEWORK_DIR to the directory containing Flutter.framework')
end

target 'Runner' do
    # use_frameworks!

    # Pods for Runner

    # Flutter Pods
    pod 'Flutter', :path => ENV['FLUTTER_FRAMEWORK_DIR']

    if File.exists? '../.flutter-plugins'
        flutter_root = File.expand_path('..')
        File.foreach('../.flutter-plugins') { |line|
            plugin = line.split(pattern='=')
            if plugin.length == 2
                name = plugin[0].strip()
                path = plugin[1].strip()
                resolved_path = File.expand_path("#{path}/ios", flutter_root)
                pod name, :path => resolved_path
                else
                puts "Invalid plugin specification: #{line}"
            end
        }
    end
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['ENABLE_BITCODE'] = 'NO'
        end
    end
end
```

**Comment by mravn-google** [comment]
Seems to be a duplicate of #15099

**Comment by NarHakobyan** [comment]
I am still receiving this error

![image](https://user-images.githubusercontent.com/1781506/67622776-c2afdc80-f82e-11e9-851f-ead2d028645d.png)

**Comment by walid-ashik** [comment]
After trying every solutions on the internet, nothing worked. I able to fix by adding the same **platform :ios, '11.0'** platform mentioned in the podfile to Build Settings and it runs and archive successfully. 

<img width="871" alt="Screen Shot 2021-08-16 at 7 57 30 AM" src="https://user-images.githubusercontent.com/33910898/129501574-3d20075b-ad2b-41b8-9783-5cfcf1ba19e1.png">
